### PR TITLE
Simplify sync logic around tipset expansions, ff and forking

### PR DESCRIPTION
**Pull request by Kubuxu**

### CODE REVIEW

File: chain/sync.go [hunk 1]
```diff
@@ -1301,31 +1301,33 @@ loop:
...
-			return blockSet, nil
-		}
+	base := blockSet[len(blockSet)-1]
+	if base.Parents() == known.Parents() {
+		// common case: receiving a block thats potentially part of the same tipset as our best block
+		return blockSet, nil
+	}
 
-		// We have now ascertained that this is *not* a 'fast forward'
+	if types.CidArrsEqual(base.Parents().Cids(), known.Cids()) {
```
Comment (schomatis) [inline chain/sync.go:1310]: This and the previous check seem to be the same now, I'd keep this one that uses the API and avoid comparing structs directly.

Comment (Kubuxu) [reply chain/sync.go:1310]: You mean keep `CidArrsEqual`?
The checks are not the same, one has one less `.Parent()`.

**Comment by Kubuxu** [comment]
@placer14 @travisperson added `chain.basefee` ^^

File: chain/sync.go (hunk 1 above)
Comment (schomatis) [reply chain/sync.go:1310]: You're right, missed that.

**Pull request merged**